Group indicator

KoderKow opened the pull request:
In reference to ([issue #1](https://github.com/elbersb/tidylog/issues/1))

After a _group_by_, filter, mutate, and transmute will have a group tag.

``` r
library(dplyr)
library(tidylog)

c <-
  iris %>%
  transmute(x = Sepal.Length + Sepal.Width)
#> transmute: dropped 5 variables (Sepal.Length, Sepal.Width, Petal.Length, Petal.Width, Species)
#> transmute: new variable 'x' with 45 unique values and 0% NA

d <-
  iris %>%
  group_by(Species) %>%
  transmute(x = Sepal.Length + Sepal.Width) %>%
  filter(x > 0) %>%
  ungroup() %>%
  filter(x < 2)
#> group_by: 3 groups (Species)
#> transmute (grouped): dropped 4 variables (Sepal.Length, Sepal.Width, Petal.Length, Petal.Width)
#> transmute (grouped): new variable 'x' with 45 unique values and 0% NA
#> filter (grouped): no rows removed
#> filter: removed all rows (100%)
```

Created on 2019-02-06 by the [reprex
package](http://reprex.tidyverse.org) (v0.2.0).

elbersb commented:
Great!

I set up Travis CI today, so pull requests will be checked -- as you can see, this one failed. Click on "Details" to see why. In this case, the linter is not happy with the single quotes in your `ifelse` statement, but wants double quotes. Could you change that? 

Also, I think it's better to use `dplyr::is.grouped_df` to check whether the data is grouped.

elbersb commented:
(You can also run the tests yourself locally before committing, of course)

KoderKow commented:
I stepping in a new world for contributing to others packages, so I greatly appreciate the help. I have fixed the quotes and also added the dplyr::is.grouped_df as you mentioned. It looks much cleaner now. I removed the blank space in " (grouped)", Travis CI pointed that out. I think it looks cleaner when it prints out "Filter(grouped):" Vs "Filter (grouped):".

> (You can also run the tests yourself locally before committing, of course)

Is this in reference to Travis CI? Before I make a pull request I can do a check with Travis CI? My apologies for not understanding that bit.

I committed the changed and the previous errors are fixed, but now the Travis CI check failed from what I can see.

elbersb commented:
Here you can see the tests that we have in the package right now: https://github.com/elbersb/tidylog/tree/master/tests/testthat

One of these runs the [lintr package](https://github.com/jimhester/lintr) on the code (for instance, it will complain when single quotes are used, or when `=` is used for assignments).

I've just added the `tidylog.Rproj` file to the repository. It's probably easiest if you open the project in Rstudio. Then you can run all the tests by pressing Shift-Cmd-T. You can read more about tests here: http://r-pkgs.had.co.nz/tests.html

Travis CI simply takes the R package and runs the tests on a clean system. But it's of course easier to make sure that the tests are okay before committing. 

Now, for your actual pull request. The linter is not happy about the trailing whitespace, i.e. whitespace at the end of the line. The linter actually didn't have a problem with the whitespace within `" {group_status}"`, and I'd prefer it with a space. Could you fix these two things?

KoderKow commented:
Thank you for this information, this helps me get up to speed on these topics. I truly appreciate your explanations! :) Moving forward I will be sure to go through the materials you presented to make this a cleaner process.

I misunderstood the trailing white space error. As you can see from my paper trail I have now reverted back to the " (grouped)" and the Travis CI build passed, but the codecov/patch did not.

elbersb commented:
Looks great! Codecov complains that the lines you added are not covered by any tests, but I'll merge the pull request now anyway.
